homeassistant/core.py

```diff
@@ -163,6 +164,11 @@ class ConfigSource(enum.StrEnum):
 
 MAX_EXPECTED_ENTITY_IDS = 16384
 
+EVENTS_EXCLUDED_FROM_MATCH_ALL = {
+    EVENT_HOMEASSISTANT_CLOSE,
+    EVENT_STATE_REPORTED,
+}
```

Comment:
When subscribing to all events, don't include EVENT_STATE_REPORTED, we require this event to be subscribed to directly.